File: base/src/main/java/com/tobi/movies/backend/Backend.java
package com.tobi.movies.backend;

import com.tobi.movies.popularstream.ApiPopularMoviesResponse;
import com.tobi.movies.posterdetails.ApiMovieDetails;

import retrofit2.http.GET;
import retrofit2.http.Path;
import rx.Observable;

public interface Backend {

    String SERVICE_ENDPOINT = "https://api.themoviedb.org/";

    @GET("/3/movie/popular")
    Observable<ApiPopularMoviesResponse> popularStream();

    @GET("/3/movie/{id}")
    Observable<ApiMovieDetails> movieDetails(@Path("id") long movieId);
}

File: base/src/main/java/com/tobi/movies/backend/Backend.kt
package com.tobi.movies.backend

import com.tobi.movies.popularstream.ApiPopularMoviesResponse
import com.tobi.movies.posterdetails.ApiMovieDetails

import retrofit2.http.GET
import retrofit2.http.Path
import rx.Observable

interface Backend {

    @GET("/3/movie/popular")
    fun popularStream(): Observable<ApiPopularMoviesResponse>

    @GET("/3/movie/{id}")
    fun movieDetails(@Path("id") movieId: Long): Observable<ApiMovieDetails>

    companion object {

        val SERVICE_ENDPOINT = "https://api.themoviedb.org/"
    }
}
